src/rc_string.c

```diff
@@ -7,4 +7,168 @@
 #include "azure_c_pal/gballoc.h"
 #include "azure_c_util/rc_string.h"
 #include "azure_c_logging/xlogging.h"
+#include "azure_c_util/thandle.h"
 
+#define THANDLE_MALLOC_FUNCTION malloc
+#define THANDLE_FREE_FUNCTION free
+THANDLE_TYPE_DEFINE(RC_STRING);
+
+#define STRING_STORAGE_TYPE_VALUES \
+    STRING_STORAGE_TYPE_COPIED, \
+    STRING_STORAGE_TYPE_MOVED, \
+    STRING_STORAGE_TYPE_WITH_CUSTOM_FREE
+
+MU_DEFINE_ENUM(STRING_STORAGE_TYPE, STRING_STORAGE_TYPE_VALUES)
+
+// In order not to expose the storage type to the user (as it is something internal to the implementation),
+// we're going to have another structure that wraps the RC_STRING and the storage type
+// It is imperative that the first member is the RC_STRING structure though, so it matches the strcuture that
```

Comment:
>strcuture [](start = 95, length = 9)

typo

Reply:
Fixed.

---
In reply to: [460251681](https://github.com/Azure/azure-c-util/pull/52#discussion_r460251681) [](ancestors = 460251681)